src/fsharp/service/IncrementalBuild.fs

```diff
@@ -391,16 +418,38 @@ type BoundModel private (tcConfig: TcConfig,
                     }
         }
 
-    member private this.TypeCheck (partialCheck: bool) =
+    member this.GetTcInfoExtras() =
+        lazyAsyncTcInfoExtras.GetValueAsync()
+
+    member this.GetTcInfoWithExtras() =
+        async {
+            match! lazyAsyncFullState.GetValueAsync() with
+            | FullState(tcInfo, tcInfoExtras) -> return tcInfo, tcInfoExtras
+            | PartialState(tcInfo) ->
+                let tcInfoExtras =
+                    {
+                        tcResolutionsRev = []
+                        tcSymbolUsesRev = []
+                        tcOpenDeclarationsRev = []
+                        latestImplFile = None
+                        itemKeyStore = None
+                        semanticClassificationKeyStore = None
+                    }
+                return tcInfo, tcInfoExtras
+        }
+
+    member private this.TypeCheck (partialCheck: bool) : Async<TcInfoState> =
         match partialCheck, lazyTcInfoState with
         | true, Some (PartialState _ as state)
-        | true, Some (FullState _ as state) -> state |> Eventually.Done
-        | false, Some (FullState _ as state) -> state |> Eventually.Done
+        | true, Some (FullState _ as state) -> async { return state }
+        | false, Some (FullState _ as state) -> async { return state }
         | _ ->
 
-        eventually {
+        asyncErrorLogger {
```

Comment:
I find it confusing to see asyncErrorLogger in some places, and async in others.   Why not asyncErrorLogger everywhere?